dbms/src/Flash/Mpp/FineGrainedShuffleWriter.cpp

```diff
@@ -57,81 +59,113 @@ void FineGrainedShuffleWriter<StreamWriterPtr>::finishWrite()
     }
 }
 
+template <class StreamWriterPtr>
+void FineGrainedShuffleWriter<StreamWriterPtr>::prepare(const Block & sample_block)
+{
+    /// Materialize sample_block so that header and reserved scatterColumns are full columns
+    /// Because ser/der don't support constant columns now
+    std::vector<Block> sample_blocks;
+    sample_blocks.push_back(sample_block);
+    HashBaseWriterHelper::materializeBlocks(sample_blocks);
+    header = sample_blocks.back().cloneEmpty();
```

Comment:
```suggestion
    header = materializeBlock(sample_block);
```
from `#include <DataStreams/materializeBlock.h>`